jvm/hermes/src/main/jni/JJSIValue.h

```diff
@@ -148,21 +152,37 @@ class JJSIValue : public JJSIValueHybridClass {
     local_ref<JJSIObject_jhybridobject> asObject(alias_ref<JRuntimeThreadContext>, alias_ref<JJSIRuntime::jhybridobject> jRuntime);
     std::string toString(alias_ref<JRuntimeThreadContext>, alias_ref<JJSIRuntime::jhybridobject> jRuntime);
 
+    ~JJSIValue() override {
+        release();
+    }
+
     void release() override {
-        value_.reset();
+        if (!tracked && value_) value_.reset();
+        if (scope_) scope_->clearRef(this);
     }
 
     bool isReleased() override {
-        return value_ == nullptr;
+        return (!tracked && value_ == nullptr) || scope_->getValue((void *)this) == nullptr;
     }
 
     Value& get_value() const {
-        if (value_) return *value_;
+        if (!tracked && value_) {
+            return *value_;
+        }
+        if (scope_) {
+            if (auto ref = scope_->getValue((void *)this)) {
+                return *ref;
+            } else {
+                throw std::runtime_error("Value reference cleaned up and is no longer in RuntimeScope");
```

Comment:
1. This should probably just go through `throwNativeHandleReleasedException("Value");` from below
2. If you don't go through the helper function, you need to tell the JNI to expect an exception, like [`throwNativeHandleReleasedException` does](https://github.com/player-ui/player/blob/dc3c3ab6fbc032b6dc44bdda553cfb5b616d93bf/jvm/hermes/src/main/jni/JJSIValue.cpp#L24-L25) (and should also probably wrap in a `PlayerException` - like `throwNativeHandleReleasedException` does)